frontend/src/pages/newPost/index.ts

```diff
@@ -43,37 +41,245 @@ const template = `
+    if (state.photos.length < 10) {
+      $fileInput.click();
+    }
+  };
+}
+
+function addPhotoHandler(fileReader: FileReader, $doneBtn: HTMLElement) {
+  return (event: Event) => {
+    const target = event.target as HTMLInputElement;
+    const file = target.files?.[0];
+    if (file) {
+      state.photos.push(file);
+      fileReader.readAsDataURL(file);
+    }
+    changeCheck($doneBtn);
+  };
+}
+
+function readFileHandler(
+  fileReader: FileReader,
+  $addPhotoBtn: HTMLElement,
+  $photoCount: HTMLElement
+) {
+  return () => {
+    const photo = {
+      src: fileReader.result?.toString() || "",
+      index: state.photos.length - 1,
+    };
+    $addPhotoBtn.insertAdjacentElement("afterend", PhotoListItem(photo));
+    renderPhotoCount($photoCount);
+  };
+}
+
+function removePhotoHandler($photoCount: HTMLElement, $doneBtn: HTMLElement) {
+  return (event: Event) => {
+    const target = event.target as HTMLElement;
+    if (!target.classList.contains("remove-btn")) {
+      return;
+    }
+
+    const $photoListItem = target.closest(".photo-list-item") as HTMLElement;
+    $photoListItem.remove();
+
+    const index = Number($photoListItem.dataset.index);
+    state.photos.splice(index, 1);
+    renderPhotoCount($photoCount);
+    changeCheck($doneBtn);
+  };
+}
+
+function renderPhotoCount($photoCount: HTMLElement) {
+  $photoCount.innerText = state.photos.length.toString();
+  if (state.photos.length) {
+    $photoCount.classList.add("active");
+  } else {
+    $photoCount.classList.remove("active");
+  }
+}
+
+function inputTitleHandler(
+  $categoryContainer: HTMLElement,
+  $doneBtn: HTMLElement
+) {
+  return (event: Event) => {
+    const target = event.target as HTMLInputElement;
+    state.title = target.value;
+    state.title
+      ? showCategories($categoryContainer)
+      : hideCategories($categoryContainer);
+    changeCheck($doneBtn);
+  };
+}
+
+function showCategories($categoryContainer: HTMLElement) {
+  $categoryContainer.classList.remove("hidden");
+}
+
+function hideCategories($categoryContainer: HTMLElement) {
+  $categoryContainer.classList.add("hidden");
+}
+
+function inputPriceHandler(event: Event) {
+  const target = event.target as HTMLInputElement;
+  const price = Number(target.value.replace(/￦| |,/g, ""));
+  if (Number.isInteger(price)) {
+    state.price = price;
+  }
+  target.value = state.price ? "￦ " + state.price.toLocaleString() : "";
+}
+
+function inputDescriptionHandler($doneBtn: HTMLElement) {
+  return (event: Event) => {
+    const target = event.target as HTMLTextAreaElement;
+    state.description = target.value;
+    changeCheck($doneBtn);
+  };
+}
+
+function changeCheck($doneBtn: HTMLElement) {
+  if (isAnyValueInvalid()) {
+    $doneBtn.classList.remove("active");
+    return;
+  }
+  $doneBtn.classList.add("active");
+}
+
+function isAnyValueInvalid(): boolean {
+  return (
+    !state.photos.length ||
+    !state.title ||
+    !state.category ||
+    !state.description
+  );
```

Comment:
```suggestion
  return !(
    state.photos.length &&
    state.title &&
    state.category &&
    state.description
  );
```

이렇게도 할 수 있을것 같네요.. 뭐가 더 보기 편할까요 ??

Reply:
`모두 yes가 아닌가?`와 `하나라도 틀렸는가?`의 차이인것 같네요 ㅋㅋㅋ
정답은 없겠지만... 제안주신 쪽이 해석하기 더 쉬운 것 같아 반영하겠습니다. 👍